### ISSUE TITLE
GM_XHR in fetch mode doesn't support nocache and revalidate options

### ISSUE THREAD

**Issue by 7nik** [Tampermonkey/tampermonkey#1003]
As I see, these options just set `Cache-Control` header.

## Expected Behavior
The header is added in both cases (prints `true`).

## Actual Behavior
The header isn't added (prints `false`).

## Specifications

Chromium: Version 83.0.4103.97 (Official Build) Arch Linux (64-bit)
TM: beta, 4.11.6116
OS: KDE Manjaro (64-bit)

## Script

```js
// ==UserScript==
// @name         Test GM_XHR-fetch nocache/revalidate
// @namespace    http://tampermonkey.net/
// @version      0.1
// @description  try to take over the world!
// @author       You
// @match        https://github.com/*
// @grant        GM_xmlhttpRequest
// @connect      httpbin.org
// ==/UserScript==

(() => {
    GM_xmlhttpRequest({
        fetch: true,
        url: "https://httpbin.org/get",
        responseType: "json",
        nocache: true,
        onload: (resp) => console.log("resp1:", resp, resp.response.headers["Cache-Control"] == "no-cache"),
        onerror: (err) => console.error("error1:", err),
    });
    GM_xmlhttpRequest({
        fetch: true,
        url: "https://httpbin.org/get",
        responseType: "json",
        revalidate: true,
        onload: (resp) => console.log("resp2:", resp, resp.response.headers["Cache-Control"] == "max-age=0, must-revalidate"),
        onerror: (err) => console.error("error2:", err),
    });
})();
```

**Comment by derjanb** [comment]
Fetch "no-store" cache mode is used in this case, which allows to control the browsers cache related behavior without setting HTTP headers: https://fetch.spec.whatwg.org/#concept-request-cache-mode

Example: please inspect the requests at TM's background page when the attached script is executed:

The first one caches the URL, 
the second one requests it with "If-None-Match: foo" and the server responds with 304 (see screenshot), 
the third and fourth requests have no "If-None-Match" header set and the server can therefore only respond with the full content and sends 200.

![image](https://user-images.githubusercontent.com/767504/87784381-7aaa6d00-c836-11ea-89d2-62a79fb1df85.png)


```JS
// ==UserScript==
// @name         Test GM_XHR-fetch nocache/revalidate
// @namespace    http://tampermonkey.net/
// @version      0.1
// @description  try to take over the world!
// @author       You
// @match        https://github.com/*
// @grant        GM_xmlhttpRequest
// @grant        GM.xmlHttpRequest
// @connect      httpbin.org
// ==/UserScript==

(() => {
    GM.xmlHttpRequest ({
        revalidate: true,
        url: "https://httpbin.org/etag/foo",
    })
    .then(() => {
        console.log("https://httpbin.org/etag/foo is cached now");
    })
    .then(() => {
        return GM.xmlHttpRequest ({
            responseType: 'json',
            url: "https://httpbin.org/etag/foo"
        })
    })
    .then(() => {
        return GM.xmlHttpRequest ({
            revalidate: true,
            nocache: true,
            responseType: 'json',
            url: "https://httpbin.org/etag/foo"
        })
    })
    .then(() => {
        return GM.xmlHttpRequest ({
            fetch: true,
            revalidate: true,
            responseType: 'json',
            url: "https://httpbin.org/etag/foo"
        })
    })
    .then(() => console.log('DONE'))
})();

```

**Comment by 7nik** [comment]
Ok `nocache` works, but what about `revalidate`?

![image](https://user-images.githubusercontent.com/29704098/87789945-7dae5900-c848-11ea-9ecd-47519b6cabaf.png)
```js
// ==UserScript==
// @name         Test GM_XHR-fetch nocache/revalidate
// @namespace    http://tampermonkey.net/
// @version      0.1
// @description  try to take over the world!
// @author       You
// @match        https://github.com/*
// @grant        GM_xmlhttpRequest
// @grant        GM.xmlHttpRequest
// @connect      cdn.donmai.us
// ==/UserScript==

(() => {
    const link = "https://cdn.donmai.us/original/eb/54/__original_drawn_by_kuroya_shinobu__eb5460253869070299d98b2d35d3cec8.png";

    GM.xmlHttpRequest ({ url: link }) // add to cache
    .then(() => GM.xmlHttpRequest ({ // force load
        nocache: true,
        url: link
    }))
    .then(() => GM.xmlHttpRequest ({ // check for changes
        revalidate: true,
        url: link
    }))
    .then(() => GM.xmlHttpRequest ({ // get from cache
        fetch: true,
        url: link
    }))
    .then(() => GM.xmlHttpRequest ({ // force load
        fetch: true,
        nocache: true,
        url: link
    }))
    .then(() => GM.xmlHttpRequest ({ // check for changes but it completely reloads
        fetch: true,
        revalidate: true,
        url: link
    }))
    .then(() => {
        console.log('DONE');
    });
})();
```

**Comment by derjanb** [comment]
Should be fixed at 4.11.6117
